src/ripple/protocol/STObject.h

```diff
@@ -538,13 +538,22 @@ class STObject
     }
 
 private:
-    void add (Serializer & s, bool withSigningFields) const;
+    enum WhichFields : bool
+    {
+        // These values are carefully chosen to do the right thing if passed
+        // to SField::shouldInclude (bool)
+        withoutSigningFields = false,
+        withAllFields = true
+    };
+
+    void add (Serializer & s, WhichFields whichFields) const;
 
     // Sort the entries in an STObject into the order that they will be
     // serialized.  Note: they are not sorted into pointer value order, they
     // are sorted by SField::fieldCode.
     static std::vector<STBase const*>
-    getSortedFields (STObject const& objToSort);
+    getSortedFields (
+        STObject const& objToSort, WhichFields whichFields);
```

Comment:
I think the only reason that method is `static` is because it does not rely on the `this` parameter.  So it's leaking an implementation detail.  I think that's not too bad since it's a private method.

Reply:
The `this` paramter should be `objToSort`.

Reply:
That signature could work, but it leaves me a little queasy about lifetime considerations.  For some reason the current signature gives me a better suggestion that the returned `vector` should not outlive the passed in `STObject&`.  But I'm not sure why I feel that way.